Update CertBuilder docs (#12386)
pyca/cryptography: docs/x509/reference.rst

    :canonical: cryptography.x509.base.CertificateBuilder

    .. versionadded:: 1.0
    
    .. note::

>        Methods such as :meth:`issuer_name` return a **new** CertificateBuilder instance
>        with the corresponding updated value. They do not modify the existing builder in place.

Reviewer: ```suggestion
       All methods, except :meth:`sign`, return a **new** CertificateBuilder
       instance with the corresponding updated value. They do not modify the
       existing builder in place.
```

        :param critical: Set to ``True`` if the extension must be understood and
             handled by whoever reads the certificate.
        

>         :return: A new :class:`CertificateBuilder` with the updated extension.

Reviewer: ```suggestion
        :return: A new :class:`CertificateBuilder` with the additional extension.
```
